src/lib.rs

```diff
@@ -53,6 +60,29 @@ pub trait TcpStack {
 	fn close(&self, socket: Self::TcpSocket) -> Result<(), Self::Error>;
 }
 
+/// This trait is implemented by TCP/IP stacks that expose TCP server functionality. TCP servers
+/// may listen for connection requests to establish multiple unique TCP connections with various
+/// clients.
+pub trait TcpServer: TcpClient {
+	/// Create a new TCP socket and bind it to the specified local port.
+	///
+	///Returns `Ok(socket)` when a new socket is successfully created and bound to the specified
+	///local port. Otherwise, an `Err(e)` variant is returned.
+	fn bind(&self, local_port: u16) -> Result<Self::TcpSocket, Self::Error>;
+
+	/// Begin listening for connection requests on a previously-bound socket.
+	///
+	/// Returns `Ok` if the socket was successfully transitioned to the listening state. Otherwise,
+	/// an `Err(e)` variant is returned.
+	fn listen(&self, socket: &mut Self::TcpSocket) -> Result<(), Self::Error>;
```

Comment:
What is the value in this extra `listen` step?  What drawbacks would there be in having `bind` go into the listening state?

Reply:
I would imagine that a user would like the ability to wait until a device is powered and operational (e.g. after all initialization) before starting to listen and accept incoming connections.

In my use cases, I'm interested in directly `listen()`-ing the socket, but I could imagine cases where a user does _not_ want that. This API is following the `berkley-sockets` model as well.